source/compiler-core/slang-artifact-util.cpp

```diff
@@ -83,6 +83,12 @@ static bool _checkRecursive(ArtifactUtil::FindStyle findStyle)
         return false;
     }
 
+    /* We currently can't write out diagnostics, so for now we'll say they are insignificant */
+    if (isDerivedFrom(desc.payload, ArtifactPayload::Diagnostics))
+    {
+        return false;
```

Comment:
Are we silently dropping diagnostics here? If so, can we alert the user by some other means?

Reply:
This means when converting the artifact hierarchy to a file system, we drop diagnostics associated with any artifacts *if* we can't serialize them. If we enable serialization (as say .txt or .json), then they will by default be included.

This doesn't change regular diagnostic reporting or anything like that. The diagnostics is still available in the "source of truth" artifact hierarchy.

Not everything associated with a artifact will be writable, and so the "isSignificant" provides a bottle neck around what we consider an error if we didn't serialize it.

It may make sense in the filtering (or in the writing to file system) to have a DiagnosticSink to report issues more broadly.